SPM Example/PortalSwift/ViewController.swift

```diff
@@ -168,308 +144,479 @@ class ViewController: UIViewController, UITextFieldDelegate {
-    self.PortalWrapper.generate { result in
-      guard result.error == nil else {
-        print("❌ handleGenerate():", result.error ?? "N/A")
-        return
-      }
-      print("✅ handleGenerate(): Address:", result.data ?? "N/A")
-      self.populateAddressInformation()
+    let privateKey = try await portal.eject(
+      withBackupMethod,
+      withCipherText: cipherTextResponse.cipherText,
+      andOrganizationBackupShare: organizationBackupShareResponse.orgShare
+    )
 
-      DispatchQueue.main.async {
-        self.dappBrowserButton?.isEnabled = true
-        self.portalConnectButton?.isEnabled = true
-        self.testButton?.isEnabled = true
-        self.signButton?.isEnabled = true
-        self.deleteKeychainButton?.isEnabled = true
-        self.testNFTsTrxsBalancesSimTrxButton?.isEnabled = true
-        self.ejectButton?.isEnabled = true
-        self.sendButton?.isEnabled = true
-        self.passkeyBackupButton?.isEnabled = true
-        self.passkeyRecoverButton?.isEnabled = true
-        self.passwordBackupButton?.isEnabled = true
-        self.passwordRecoverButton?.isEnabled = true
-        self.gdriveBackupButton?.isEnabled = true
-        self.gdriveRecoverButton?.isEnabled = true
-        self.iCloudBackupButton?.isEnabled = true
-        self.iCloudRecoverButton?.isEnabled = true
-      }
+    return privateKey
+  }
+
+  public func generate() async throws -> PortalCreateWalletResponse {
+    guard let portal else {
+      self.logger.error("PortalWrapper.generate() - Portal not initialized. Please call registerPortal().")
+      throw PortalExampleAppError.portalNotInitialized()
     }
+
+    return try await portal.createWallet()
   }
 
-  // Call this function whenever you want to prompt the user for a PIN
-  func requestPassword(completion: @escaping (String?) -> Void) {
-    let alertController = UIAlertController(title: "Enter Password", message: nil, preferredStyle: .alert)
+  func getBalances() async throws -> Bool {
+    guard let portal else {
+      throw PortalExampleAppError.portalNotInitialized()
+    }
+    let chainId = "eip155:11155111"
+    let balances = try await portal.getBalances(chainId)
+
+    return true
+  }
 
-    // Add text field for PIN input
-    alertController.addTextField { textField in
-      textField.placeholder = "PASSWORD"
-      textField.isSecureTextEntry = true
-      textField.keyboardType = .numberPad
+  public func getGasPrice(_ chainId: String) async throws {
+    guard let portal else {
+      throw PortalExampleAppError.portalNotInitialized()
     }
 
-    // Submit action
-    let submitAction = UIAlertAction(title: "Submit", style: .default) { _ in
-      let password = alertController.textFields?.first?.text
-      completion(password)
+    let gasPriceResponse = try await portal.request(chainId, withMethod: .eth_gasPrice, andParams: [])
+
+    print("Gas price response: \(gasPriceResponse)")
+
+//    return gasPriceResponse.result
```

Comment:
Can probably remove this since we only care about the log 🙂

Reply:
I left it since it's just in the example app and we're likely to add it again at some point in the future :)